src/StateSpace.hpp

```diff
@@ -28,6 +28,58 @@ class StateSpace {
     virtual T intermediateState(const T &source, const T &target, float stepSize) const = 0;
 
     /**
+     * An overloaded version designed for use in adaptive stepsize control. A previous
+     * stepsize of 0 will indicate that the this is the first iteration of the tree and will
+     * automatically set the stepsize to defaultStepSize.
+     *
+     * @param source The node in the tree to extend from
+     * @param target The point in the space to extend to
+     * @param prevStepSize The stepsize taken to reach source
+     * @param ascGrowthRate How aggressively the RRT grows or shrinks
+     * @param defaultStepSize The default stepsize of the RRT\
+     *
+     * @return A state in the direction of @target from @source.state()
+     */
+    virtual T intermediateState(const T &source, const T &target, float prevStepSize, float ascGrowthRate, float defaultStepSize) const = 0;
+
+    /**
+     * Sets the minimum possible stepsize the tree can extend when running adaptive stepsize control.
+     *
+     * @param minStepSize The minimum stepsize.
+     */
+    virtual void setMinStepSize(float minStepSize) = 0;
```

Comment:
I think it might make more sense to put these properties on `RRT:Tree` rather than `StateSpace`.  The `StateSpace` class is meant to encapsulate details about the environment being explored, while the `Tree` class holds the details about *how* it will be explored.  Also, the `stepSize` property is already on `Tree`, so it would be better if `maxStepSize` was there too.